### DISCUSSION TITLE
how to add object to array

### DISCUSSION THREAD

**Discussion by antoniopetricc** [nlohmann/json#2768] [Q&A] [answered]
i have the following structure:
 
`{
    "classes": [
        {
            "title": "1B",
            "students": [
                "Tony",
                "Pablo"
            ]
        }
    ]
}`

how can i add another object to the classes array?

like this:

`{
    "classes": [
        {
            "title": "1B",
            "students": [
                "Tony",
                "Pablo"
            ]
        },
        {
            "title": "2B",
            "students": [
                "Pier",
                "Gianfranpaolo"
            ]
        }
    ]
}`

**Comment by nlohmann** [comment, marked as answer]
Assuming you parsed the JSON to a `json` object called `j`, then you can do

```cpp
json new_element;
new_element["title"] = "2B";
new_element["students"].push_back("Pier");
new_element["students"].push_back("Gianfranpaolo");

j["classes].push_back(std::move(new_element));
```
